kernel/src/engine/arrow_expression.rs

```diff
@@ -175,14 +193,18 @@ fn column_as_struct<'a>(
     name: &str,
     column: &Option<&'a Arc<dyn Array>>,
 ) -> Result<&'a StructArray, ArrowError> {
-    column
-        .ok_or(ArrowError::SchemaError(format!("No such column: {}", name)))?
-        .as_any()
-        .downcast_ref::<StructArray>()
-        .ok_or(ArrowError::SchemaError(format!("{} is not a struct", name)))
+    let c = column.ok_or(ArrowError::SchemaError(format!("No such column: {}", name)))?;
+    if let arrow_schema::DataType::Map(_, _) = c.data_type() {
```

Comment:
nit: I recently learned that `..` syntax works here as well, so you don't have to care about the number of values:
```suggestion
    if let arrow_schema::DataType::Map(..) = c.data_type() {
```